File: frontend/app/src/main/java/com/h1oo7/musicapp/ui/fragment/ProfileFragment.java
package com.h1oo7.musicapp.ui.fragment;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.h1oo7.musicapp.ui.activity.LoginActivity;
import com.h1oo7.musicapp.R;
import com.h1oo7.musicapp.utils.SharedPrefManager;

import de.hdodenhof.circleimageview.CircleImageView;

public class ProfileFragment extends Fragment {

    private CircleImageView imgAvatar;
    private TextView tvDisplayName, tvUsername, tvRole, tvUserId;
    private Button btnLogout;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_profile, container, false);

        imgAvatar = view.findViewById(R.id.img_avatar);
        tvDisplayName = view.findViewById(R.id.tv_display_name);
        tvUsername = view.findViewById(R.id.tv_username);
        tvRole = view.findViewById(R.id.tv_role);
        tvUserId = view.findViewById(R.id.tv_user_id);
        btnLogout = view.findViewById(R.id.btn_logout);

        loadUserInfo();

        btnLogout.setOnClickListener(v -> logout());

        return view;
    }

    private void loadUserInfo() {
        SharedPrefManager pref = SharedPrefManager.getInstance(requireContext());

        tvDisplayName.setText(pref.getDisplayName() != null ? pref.getDisplayName() : "Người dùng");
        tvUsername.setText("@" + (pref.getUsername() != null ? pref.getUsername() : "user"));
        tvUserId.setText("ID: " + (pref.getUserId() != null ? pref.getUserId() : "N/A"));

        String role = pref.getUserRole();
        tvRole.setText("Vai trò: " + (role != null && role.equals("admin") ? "Quản trị viên" : "Người dùng"));
    }

    private void logout() {
        SharedPrefManager.getInstance(requireContext()).logout();
        Intent intent = new Intent(requireActivity(), LoginActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        startActivity(intent);
        requireActivity().finish();
    }
}

File: forntend/app/src/main/java/com/example/musique/ui/fragments/ProfileFragment.kt
package com.example.musique.ui.fragments

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import com.example.musique.data.network.RetrofitClient
import com.example.musique.data.repository.AuthRepository
import com.example.musique.databinding.FragmentProfileBinding
import com.example.musique.ui.activities.LoginActivity
import com.example.musique.ui.viewmodel.AuthViewModel
import com.example.musique.ui.viewmodel.ViewModelFactory
import com.example.musique.utils.PreferenceManager

class ProfileFragment : Fragment() {
    
    private var _binding: FragmentProfileBinding? = null
    private val binding get() = _binding!!
    
    private lateinit var viewModel: AuthViewModel
    private lateinit var preferenceManager: PreferenceManager
    
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentProfileBinding.inflate(inflater, container, false)
        return binding.root
    }
    
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        
        setupViewModel()
        loadUserData()
        setupClickListeners()
    }
    
    private fun setupViewModel() {
        preferenceManager = PreferenceManager(requireContext())
        val apiService = RetrofitClient.getApiService(preferenceManager)
        val repository = AuthRepository(apiService, preferenceManager)
        val factory = ViewModelFactory(authRepository = repository)
        viewModel = ViewModelProvider(this, factory)[AuthViewModel::class.java]
    }
    
    private fun loadUserData() {
        binding.tvDisplayName.text = preferenceManager.getDisplayName()
        binding.tvUsername.text = "@${preferenceManager.getUsername()}"
    }
    
    private fun setupClickListeners() {
        binding.btnLogout.setOnClickListener {
            viewModel.logout()
            navigateToLogin()
        }
    }
    
    private fun navigateToLogin() {
        val intent = Intent(requireContext(), LoginActivity::class.java)
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
        startActivity(intent)
        requireActivity().finish()
    }
    
    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
